server/src/main/java/org/candlepin/resource/OwnerProductResource.java

```diff
@@ -458,10 +463,10 @@ public void deleteProduct(
     @Produces(MediaType.APPLICATION_JSON)
     @Consumes(MediaType.WILDCARD)
     @Transactional
-    public JobDetail refreshPoolsForProduct(
+    public AsyncJobStatusDTO refreshPoolsForProduct(
         @PathParam("owner_key") String ownerKey,
         @PathParam("product_id") String productId,
-        @QueryParam("lazy_regen") @DefaultValue("true") Boolean lazyRegen) {
+        @QueryParam("lazy_regen") @DefaultValue("true") boolean lazyRegen) throws JobException {
```

Comment:
Not sure it is a good idea to change to primitive here. I think we're using the wrappers for some reason (which I can't remember right now). @Ceiu might know.

Reply:
I switched it to primitive since it has a default value so it is always present.

Reply:
I think it might have to do with being able to explicitly send a null value for that field, and using a primitive might fail doing that (I've not verified this though).

Reply:
Yeah, I saw this in the review and thought what @Januson stated -- with a default value, it should be fine as a primitive. Admitted, I didn't consider trying to pass in null, but I don't think it's possible to do so via REST. I believe any value will be considered a string and converted accordingly.